modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -18,111 +18,133 @@ InputParameters validParams<MultiSmoothSuperellipsoidIC>()
-    _semiaxis_a(getParam<Real>("semiaxis_a")),
-    _semiaxis_b(getParam<Real>("semiaxis_b")),
-    _semiaxis_c(getParam<Real>("semiaxis_c")),
-    _semiaxis_a_variation(getParam<Real>("semiaxis_a_variation")),
-    _semiaxis_b_variation(getParam<Real>("semiaxis_b_variation")),
-    _semiaxis_c_variation(getParam<Real>("semiaxis_c_variation")),
-    _semiaxis_variation_type(getParam<MooseEnum>("semiaxis_variation_type"))
+    _semiaxis_variation_type(getParam<MooseEnum>("semiaxis_variation_type")),
+    _prevent_overlap(getParam<MooseEnum>("prevent_overlap")),
+    _check_extremes(getParam<MooseEnum>("check_extremes")),
+    _numbub_vec(parameters.get<std::vector<unsigned int> >("numbub")),
+    _bubspac_vec(parameters.get<std::vector<Real> >("bubspac")),
+    _exponent_vec(parameters.get<std::vector<Real> >("exponent")),
+    _semiaxis_a_vec(parameters.get<std::vector<Real> >("semiaxis_a")),
+    _semiaxis_b_vec(parameters.get<std::vector<Real> >("semiaxis_b")),
+    _semiaxis_c_vec(parameters.get<std::vector<Real> >("semiaxis_c")),
+    _semiaxis_a_variation_vec(parameters.get<std::vector<Real> >("semiaxis_a_variation")),
+    _semiaxis_b_variation_vec(parameters.get<std::vector<Real> >("semiaxis_b_variation")),
+    _semiaxis_c_variation_vec(parameters.get<std::vector<Real> >("semiaxis_c_variation"))
 {
 }
 
 void
 MultiSmoothSuperellipsoidIC::initialSetup()
 {
+  _nv = _numbub_vec.size();
 
-  //Set up domain bounds with mesh tools
-  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+  if (_nv != _bubspac_vec.size() || _nv != _exponent_vec.size() || _nv != _semiaxis_a_vec.size() || _nv != _semiaxis_b_vec.size()|| _nv != _semiaxis_c_vec.size())
+    mooseError("Vectors for numbub, bubspac, exponent, semiaxis_a, semiaxis_b, and semiaxis_c must be the same size.");
+
+  if (_semiaxis_variation_type != 2)
   {
-    _bottom_left(i) = _mesh.getMinInDimension(i);
-    _top_right(i) = _mesh.getMaxInDimension(i);
+    if (_nv != _semiaxis_a_variation_vec.size() || _nv != _semiaxis_b_variation_vec.size() || _nv != _semiaxis_c_variation_vec.size())
+      mooseError("Vectors for numbub, semiaxis_a_variation, semiaxis_b_variation, and semiaxis_c_variation must be the same size.");
   }
-  _range = _top_right - _bottom_left;
 
-  if (_semiaxis_a_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_a_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
-  if (_semiaxis_b_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_b_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
-  if (_semiaxis_c_variation > 0.0 && _semiaxis_variation_type == 2)
-    mooseError("If Semiaxis_c_variation > 0.0, you must pass in a Semiaxis_variation_type in MultiSmoothSuperellipsoidIC");
+  for (unsigned int k = 0; k < _nv; k++)
+  {  //Set up domain bounds with mesh tools
+    _numbub = _numbub_vec.at(k);
```

Comment:
You're using `vector::at()` which does range-checking and throws an exception on out-of-range access.  However you are not actually catching this potential exception anywhere... Therefore it's equivalent (and more efficient, but I don't really care about that as much as the code uniformity aspect) to just use the normal `vector::operator[]`.